params/hooks_libevm.go

```diff
@@ -0,0 +1,246 @@
+	return addresses
+}
+
+// precompileOverrideBuiltin specifies precompiles that were activated prior to the
+// dynamic precompile activation registry.
+// These were only active historically and are not active in the current network.
+func (r RulesExtra) precompileOverrideBuiltin(addr common.Address) (libevm.PrecompiledContract, bool) {
+	var precompiles map[common.Address]contract.StatefulPrecompiledContract
+	switch {
+	case r.IsBanff:
+		precompiles = PrecompiledContractsBanff
+	case r.IsApricotPhase6:
+		precompiles = PrecompiledContractsApricotPhase6
+	case r.IsApricotPhasePre6:
+		precompiles = PrecompiledContractsApricotPhasePre6
+	case r.IsApricotPhase2:
+		precompiles = PrecompiledContractsApricotPhase2
+	}
+
+	precompile, ok := precompiles[addr]
+	if !ok {
+		return nil, false
+	}
+
+	return makePrecompile(precompile), true
+}
+
+func makePrecompile(contract contract.StatefulPrecompiledContract) libevm.PrecompiledContract {
+	run := func(env vm.PrecompileEnvironment, input []byte, suppliedGas uint64) ([]byte, uint64, error) {
+		header, err := env.BlockHeader()
+		if err != nil {
+			panic(err) // Should never happen
+		}
+		var predicateResultsBytes []byte
+		if len(header.Extra) >= DynamicFeeExtraDataSize {
+			predicateResultsBytes = header.Extra[DynamicFeeExtraDataSize:]
+		}
+		accessableState := accessableState{
+			env: env,
+			blockContext: &BlockContext{
+				number:                env.BlockNumber(),
+				time:                  env.BlockTime(),
+				predicateResultsBytes: predicateResultsBytes,
+			},
+		}
+		return contract.Run(accessableState, env.Addresses().Caller, env.Addresses().Self, input, suppliedGas, env.ReadOnly())
+	}
+	return vm.NewStatefulPrecompile(run)
+}
+
+func (r RulesExtra) PrecompileOverride(addr common.Address) (libevm.PrecompiledContract, bool) {
+	if p, ok := r.precompileOverrideBuiltin(addr); ok {
+		return p, true
+	}
+	if _, ok := r.Precompiles[addr]; !ok {
+		return nil, false
+	}
+	module, ok := modules.GetPrecompileModuleByAddress(addr)
+	if !ok {
+		return nil, false
+	}
+
+	return makePrecompile(module.Contract), true
+}
+
+type accessableState struct {
+	env          vm.PrecompileEnvironment
+	blockContext *BlockContext
+}
+
+func (a accessableState) GetStateDB() contract.StateDB {
+	// XXX: this should be moved to the precompiles
+	var state libevm.StateReader
+	if a.env.ReadOnly() {
+		state = a.env.ReadOnlyState()
+	} else {
+		state = a.env.StateDB()
+	}
+	return state.(contract.StateDB)
+}
+
+func (a accessableState) GetBlockContext() contract.BlockContext {
+	return a.blockContext
+}
+
+func (a accessableState) GetChainConfig() precompileconfig.ChainConfig {
+	return GetExtra(a.env.ChainConfig())
+}
+
+func (a accessableState) GetSnowContext() *snow.Context {
+	return GetExtra(a.env.ChainConfig()).SnowCtx
+}
+
+func (a accessableState) NativeAssetCall(caller common.Address, input []byte, suppliedGas uint64, gasCost uint64, readOnly bool) (ret []byte, remainingGas uint64, err error) {
```

Comment:
yes I moved this but to avoid a circular import we would have to either pass the CallNewAccountGas through the precompile (which I did here) or to import that from gethparams